core/query/sql.ml

```diff
@@ -137,138 +152,162 @@ let order_by_clause n =
+  let pr_delete ppf table where =
+    let pp_where ppf where = gen_pp_option ppf "where (%a)" pr_b_ignore_fields where in
+    Format.fprintf ppf "delete from %a%a" pp_constant table pp_where where
   in
-  let string_of_insert table fields values =
-    let fields = String.concat ", " fields in
-    let values =
-      values
-      (* Concatenate and bracket the values in each row *)
-      |> List.map ((List.map sbt) ->- String.concat ", " ->- Printf.sprintf "(%s)")
-      (* Join all rows *)
-      |> String.concat ", " in
-    Printf.sprintf "insert into %s (%s) values %s"
-      table fields values
+  let pr_update ppf table fields where =
+    let pp_field ppf (k, v) = gen_pp_pair ppf "%a = %a" pp_quote pr_b_ignore_fields (k, v) in
+    let pp_where ppf where = gen_pp_option ppf "where (%a)" pr_b_ignore_fields where in
+    Format.fprintf ppf "update %a set %a %a" pp_constant table (Format.pp_print_list ~pp_sep:pp_comma pp_field) fields pp_where where
   in
-    match q with
-      | UnionAll ([], _) -> "select 42 as \"@unit@\" where false"
-      | UnionAll ([q], n) -> sq q ^ order_by_clause n
-      | UnionAll (qs, n) ->
-        mapstrcat " union all " (fun q -> "(" ^ sq q ^ ")") qs ^ order_by_clause n
-      | Select (fields, [], Constant (Constant.Bool true), _os) ->
-          let fields = string_of_fields fields in
-            "select " ^ fields
-      | Select (fields, [], condition, _os) ->
-          let fields = string_of_fields fields in
-            "select * from (select " ^ fields ^ ") as " ^ fresh_dummy_var () ^ " where " ^ sb condition
-      | Select (fields, tables, condition, os) ->
-          (* using quote_field assumes tables contains table names (not nested queries) *)
-          let tables = List.map (fun (t, x) -> quote t ^ " as " ^ (string_of_table_var x)) tables
-          in string_of_select fields tables condition os
-      | Delete { del_table; del_where } ->
-          string_of_delete del_table del_where
-      | Update { upd_table; upd_fields; upd_where } ->
-          string_of_update upd_table upd_fields upd_where
-      | Insert { ins_table; ins_fields; ins_records } ->
-          string_of_insert ins_table ins_fields ins_records
-      | With (_, q, z, q') ->
-          match q' with
-          | Select (fields, tables, condition, os) ->
-              (* Inline the query *)
-              let tables = List.map (fun (t, x) -> quote t ^ " as " ^ (string_of_table_var x)) tables in
-              let q = "(" ^ sq q ^ ") as " ^ string_of_table_var z in
-              string_of_select fields (q::tables) condition os
-          | _ -> assert false
+  let pr_insert ppf table fields values =
+    let pp_value ppf x = Format.fprintf ppf "(%a)" (Format.pp_print_list ~pp_sep:pp_comma pr_b_ignore_fields) x in
+    Format.fprintf ppf "insert into %a (%a) values %a"
+      pp_constant table
+      (Format.pp_print_list ~pp_sep:pp_comma Format.pp_print_string) fields
```

Comment:
use pp_constant / pp_string here instead of Format.pp_print_string for consistency?